scripts/release/version_doc.py

```diff
@@ -0,0 +1,34 @@
+import yaml
+
+
+def table(list_of_dicts):
+    markdown_table = ""
+    markdown_header = '| ' + ' | '.join(map(str, list_of_dicts[0].keys())) + ' |'
+    markdown_header_separator = '|-----' * len(list_of_dicts[0].keys()) + '|'
+    markdown_table += markdown_header + '\n'
+    markdown_table += markdown_header_separator + '\n'
+    for row in list_of_dicts:
+        markdown_row = ""
+        for key, col in row.items():
+            markdown_row += '| ' + str(col) + ' '
+        markdown_row += '|' + '\n'
+        markdown_table += markdown_row
+    return markdown_table
+
+
+def version_doc(args):
+    input_file = args.input_file
+    out_file = args.out_file
+    with open(input_file, 'r') as f:
+        versions = yaml.safe_load(f)
+
+    with open('template/version_doc.md', 'r') as vd:
+        final_md = vd.read()
+
+    table_md = table(versions)
+
+    final_md = final_md.replace('<<TABLE>>', table_md)
```

Comment:
If we're going to use a magic keyword we should be a little more clear here on what `<<TABLE>>` is (rename to something like `<<GENERATED_COMPATIBILITY_TABLE>>`)

Reply:
it's probably not worth it here for a single string-replace, but if we ever wanted to expand capabilities here a better pattern here would to use a templater like jinja fwiw